web/src/main/java/org/springframework/security/web/reactive/result/method/annotation/AuthenticationPrincipalArgumentResolver.java

```diff
@@ -69,9 +70,9 @@ public boolean supportsParameter(MethodParameter parameter) {
 	public Mono<Object> resolveArgument(MethodParameter parameter, BindingContext bindingContext,
 			ServerWebExchange exchange) {
 		ReactiveAdapter adapter = getAdapterRegistry().getAdapter(parameter.getParameterType());
-		return exchange.getPrincipal()
-			.ofType(Authentication.class)
-			.flatMap( a -> {
+		return ReactiveSecurityContextHolder.getContext()
+			.map(SecurityContext::getAuthentication)
```

Comment:
@Dav1dde  @eleftherias 
By javadoc `SecurityContext::getAuthentication` could return `null`  
`Mono.map` forbids `null`  
should this case be handled, or it is very unlikely to happen

Reply:
Looking at the implementation of `ReactiveSecurityContextHolder` this may very well be the case (the API allows it).

This would also need to be fixed in multiple other `AuthenticationPrinicipalArgumentResolver` implementations, e.g. in [messaging](https://github.com/spring-projects/spring-security/blob/master/messaging/src/main/java/org/springframework/security/messaging/handler/invocation/reactive/AuthenticationPrincipalArgumentResolver.java#L129)